src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -168,45 +181,66 @@ public boolean isBound() {
    * @throws Exception if closing fails server-side.
    */
   public void close() throws Exception {
-    if (this.getStatementResult() != null) {
-      this.getStatementResult().close();
+    for (int index = 0; index < this.getStatementCount(); index++) {
+      close(index);
+    }
+  }
+
+  public void close(int index) throws Exception {
+    if (this.getStatementResult(index) != null) {
+      this.getStatementResult(index).close();
     }
   }
 
   /** @return True if this is a select statement, false otherwise. */
-  public boolean containsResultSet() {
-    return this.resultType == ResultType.RESULT_SET;
+  public boolean containsResultSet(int index) {
+    return this.statementTypes.get(index) == StatementType.QUERY;
   }
 
   /** @return True if this statement was executed, False otherwise. */
   public boolean isExecuted() {
-    return executed;
+    return lastExecutedIndex >= 0;
+  }
+
+  public int getLastExecutedIndex() {
+    return lastExecutedIndex;
   }
 
   /** @return The number of items that were modified by this execution. */
-  public Long getUpdateCount() {
-    return this.updateCount;
+  public long getUpdateCount(int index) {
+    if (this.updateCounts == null) {
+      return 0;
+    }
+    return this.updateCounts[index];
```

Comment:
This will not be exposed to customers through psql or pg drivers. For internal implementation, it is fine because we can know ahead that this is a query/ddl/client_side through getStatementTypes() and we never call this method in those cases (except for testing).

Reply:
As an alternative, I can return -1, which complies to what users will observe through jdbc's getUpdateCount()

Reply:
If this is guaranteed not to be returned to clients when it is not the result, then I'm fine with this being 0 (zero). I think it would be good to add it to the documentation of the method that the result is only valid if the statement type at the given index is `UPDATE`.

Reply:
Done.